### PR TITLE
Fix issue #10366

### PR DESCRIPTION
**Pull request by kavishkafernando**
Fixes #https://github.com/wso2/product-apim/issues/10366

### CODE REVIEW

**Review by tmkasun — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx [hunk 1]
```diff
@@ -283,6 +284,38 @@ function EndpointOverview(props) {
...
+                endpointCategory.sandbox
+                && endpointConfig.sandbox_endpoints
+                && endpointConfig.sandbox_endpoints && endpointConfig.sandbox_endpoints.url !== ''
+                && !endpointCategory.production
+                && !endpointConfig.production_endpoints
+            ) {
+                setIsEndpointUrlAvailable(true);
+            } else {
+                setIsEndpointUrlAvailable(false);
+            }
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx:315]: Shall we simplify this a bit: 
Example:
```suggestion

            var productionEpSpecified =  endpointCategory.prod
                        && endpointConfig.production_endpoints
                        && endpointConfig.production_endpoints.url && endpointConfig.production_endpoints.url !== '';
            var sandboxEpSpecified =  endpointCategory.sandbox
                && endpointConfig.sandbox_endpoints
                && endpointConfig.sandbox_endpoints.url && endpointConfig.sandbox_endpoints.url !== ''
            var productionEpNotSelected = !endpointCategory.sandbox && !endpointConfig.sandbox_endpoints;
            var sandboxEpNotSelected = !endpointCategory.sandbox && !endpointConfig.sandbox_endpoints;
            
            if ((productionEpSpecified && sandboxEpSpecified)
                 || (productionEpSpecified && sandboxEpNotSelected)
                 || (sandboxEpSpecified && productionEpNotSelected)) {
                setIsEndpointUrlAvailable(true);
            } else {
                setIsEndpointUrlAvailable(false);
            }
```

Will this work?

Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx [hunk 2]
```diff
@@ -283,6 +284,30 @@ function EndpointOverview(props) {
         }
     }, [props]);
 
+    useEffect(() => {
+        const prodEpSpecified = endpointCategory.prod
+            && endpointConfig.production_endpoints
+            && endpointConfig.production_endpoints
+            && endpointConfig.production_endpoints.url !== '';
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx:291]:
```suggestion
            && endpointConfig.production_endpoints
            && endpointConfig.production_endpoints.url !== '';
```

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx [hunk 3]
```diff
@@ -283,6 +284,30 @@ function EndpointOverview(props) {
...
+        const sandboxEptNotSpecified = !endpointCategory.sandbox && !endpointConfig.sandbox_endpoints;
+        const sandboxEpSpecified = endpointCategory.sandbox
+            && endpointConfig.sandbox_endpoints
+            && endpointConfig.sandbox_endpoints
+            && endpointConfig.sandbox_endpoints.url !== '';
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx:296]:
```suggestion
            && endpointConfig.sandbox_endpoints
            && endpointConfig.sandbox_endpoints.url !== '';
```

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**